### DISCUSSION TITLE
Inbox Improvments

### DISCUSSION THREAD

**Discussion by liefygreen** [AprilSylph/XKit-Rewritten#229] [Suggestion Box]
Hello, one thing that I would really like is some organization for the inbox. For example, to filter my ask messages to just mutuals, just followers, or just anons (or to filter out anons). 

I also think it would be great if there was a way to do blacklisting in your inbox. So, for example, you can blacklist certain words and block out any asks that contain those words, because that would help anyone who wanted to keep anon open, but was receiving harassment and hate messages. Maybe an option to choose between either hiding the existence of those asks entirely, or just having them be blacked out in the inbox with the option to reveal (so then if the asker isn't on anon, you can look at the username and make a decision whether or not you trusted the sender before actually reading the ask).

**Comment by AprilSylph** [comment]
Does content filtering not catch terms in your own inboxes? I would [contact Tumblr support](https://www.tumblr.com/support) to ask if that's intentional.

**Reply by liefygreen** [reply]
I just tested it and no, content filtering does not catch terms in the inbox (and from the description of the blacklisting function, that seems intentional?). I'll bring it up to tumblr support, though.
Regardless, in my mind, there should be a separate filter in the inbox than from the rest of tumblr. So, for example, I am a bi man and a lot of my mutuals are mlm who reclaim the f slur and use it in jokes and to refer to themselves. I'm fine with seeing posts where they do that.

But say some people get mad with me for whatever reason and started sending me homophobic anons. In that case, I would want to filter words like the f slur in some way exclusively in my inbox. I still want to see the posts on my dash from my mutuals and the blogs I follow that contain that word, because it's not hurtful in that context. I also don't want to close off my asks to my followers and mutuals just because some people were being awful. So the best thing is just to have a way of filtering that word and certain terms specifically within my inbox itself.

Even if they made blacklist extend to the inbox, itself, I would still want the ability to just have a filter exclusive for the inbox. Especially if it's for a term like "trans" or "gender" for example and I was getting transphobic hate. I don't want to filter the words "trans" and "gender" and have to click through to see every post on my dash that contains the word "trans" or "gender". That would be like, 1/3 of my dashboard. Especially since tumblr's blacklist function looks at usernames as well as post content.

I'm only really worried about the way that certain words are being used when directed at me in an ask. The current blacklist function is more about avoiding things that are triggering or upsetting in any context. No need to filter something across all of tumblr when you are only worried about the way it may be used in your inbox.
